zerver/actions/user_status.py

```diff
@@ -51,4 +56,31 @@ def do_update_user_status(
         event["emoji_name"] = emoji_name
         event["emoji_code"] = emoji_code
         event["reaction_type"] = reaction_type
+
+    if scheduled_end_time is not None:
+        event["scheduled_end_time"] = scheduled_end_time
+
     send_event(realm, event, get_user_ids_who_can_access_user(user_profile))
+
```

Comment:
I just realised I am sending the `scheduled_end_time` event to everyone, rather than just the user who set it.

Reply:
What if we are only updating the `scheduled_end_time`, in that case the later event is unnecessary actually.

Reply:
the problem with `user_status_ui` is for change in one of the field values, all the other field values are also populated in the request. So I guess to remove that extra event, I guess the modal should be updated to populate the request with only fields that are updated.?

Reply:
@prakhar1144 I don't think it is possible to do that without querying to check if only the `scheduled_end_time` has changed. The request can't contain just the `scheduled_end_time` in your case because then it would open the possibility of setting a `scheduled_end_time` without a user status for a user.

Reply:
What I meant to say is: A user updates their `scheduled_end_time` time. So, if my status is *In a meeting* and end time 1hr then I realized the meeting is going to take one more hour, so I will just update the end time. -- It is a very practical case, right ?

The approach should be:
1.  Update this block in `submit_new_status` function to include `scheduled_end_time` too. i.e no POST request if neither of the field was changed.
```
    if (
        old_status_text === new_status_text &&
        !emoji_status_fields_changed(selected_emoji_info, old_emoji_info)
    ) {
        dialog_widget.close();
        return;
    }
```
2. Include `scheduled_end_time` as a `UserStatusData` field in `user_status_ui.ts`
3. No need of `if (scheduled_end_time !== undefined) {` check in `submit_new_status` function. -- simply call `server_update_status` with `...data`.
4. Update the following block in `update_user_status_backend()` to include `scheduled_end_time` -- Also in this view we can add that check for "it would open the possibility of setting a scheduled_end_time without a user status for a user."
```
    if (away is None) and (status_text is None) and (emoji_name is None):
        raise JsonableError(_("Client did not pass any new values."))
 ```
 These checks make sure that atleast one field is getting updated.
 
 5. Now, there are two approaches I can think of:
 
 A.  before sending event decide what should be the value of `user_ids_to_send_event` 
 ```
 if only_scheduled_end_time_updated:
     user_ids_to_send_event = [user_profile.id]
 else:
     user_ids_to_send_event = get_user_ids_who_can_access_user(user_profile)
     
 send_event(realm, event, user_ids_to_send_event)
 ```
 We can determine `only_scheduled_end_time_updated` using db query. This will save unnecessary events sent to other users when only end time is updated. 
 
 B. Send event to everyone and update the end_time only for the user with user_id event["user_id"] (When only end_time is updated) -- This will save db query but unnecessary events will get sent to other users in that case.

Reply:
updated!